### PR TITLE
Nodejs binding and demo for Rhino v2.0

### PR DESCRIPTION
**Pull request by mrrostam**
@laves can you do me a favour and check the mac x86 lib file specifically as I built it on my m1 :)

### CODE REVIEW

**Comment by ErisMik** [comment]
FYI @mrrostam You can build the Mac `x86` binary by opening up iterm / terminal in Rosetta and building it as you normally would. It will still output the build into the `arm64` directory but the libs will be for `x86`. You can verify using `file`.

File: binding/nodejs/platforms.js [hunk 1]
```diff
@@ -141,6 +202,13 @@ function getSystemLibraryPath() {
           }
         }
       }
+      case SYSTEM_WINDOWS: {
+        if (arch == X86_64) {
```
Comment (ErisMik) [inline binding/nodejs/platforms.js:206]: `===` to make it more consistent

File: binding/nodejs/platforms.js [hunk 2]
```diff
@@ -121,6 +178,10 @@ function getSystemLibraryPath() {
           return absoluteLibraryPath(
             SYSTEM_TO_LIBRARY_PATH.get(`${SYSTEM_MAC}/${X86_64}`)
           );
+        } else if (arch == ARM_64) {
```
Comment (ErisMik) [inline binding/nodejs/platforms.js:181]: `===` to make it more consistent

**Review by ErisMik — APPROVED** [review]

File: binding/nodejs/index.js [hunk 3]
```diff
@@ -41,11 +41,20 @@ class Rhino {
    * @param {string} manualLibraryPath the path to the Rhino dynamic library (platform-dependent extension)
    */
   constructor(
+    accessKey,
```
Comment (laves) [inline binding/nodejs/index.js:44]: what about `requires_endpoint`? is that not part of the Rhino node API?

File: demo/nodejs/mic.js [hunk 4]
```diff
@@ -18,6 +18,10 @@ const { PvArgumentError } = require("@picovoice/rhino-node/errors");
 const PvRecorder = require("@picovoice/pvrecorder-node");
 
 program
+  .requiredOption(
+    "-a, --access_key <string>",
```
Comment (laves) [inline demo/nodejs/mic.js:22]: `requires_endpoint`?

**Review by laves — CHANGES_REQUESTED** [review]

**Comment by laves** [comment]
I already built the node binaries as part of my big lib update, you probably don't need to include those in the merge

**Pull request closed without merging**